index.ts

```diff
@@ -12,6 +12,18 @@ type Attributes = JSX.IntrinsicElements['div'];
 type DocumentFragmentConstructor = typeof DocumentFragment;
 type ElementFunction = () => HTMLElement | SVGElement;
 
+declare global {
+	namespace JSX {
+		interface Element extends HTMLElement {}
+	}
+}
+
+interface JSXElementClassDocumentFragment extends DocumentFragment, JSX.ElementClass {}
+interface Fragment {
+	prototype: JSXElementClassDocumentFragment;
+	new (): JSXElementClassDocumentFragment;
+}
```

Comment:
I'm not sure, but isn't this equivalent to
```suggestion
type Fragment = JSXElementClassDocumentFragment;
```

Reply:
no its not equivalent because you need the `prototype` and `new ()` props to be `JSXElementClassDocumentFragment`, otherwise you get errors.